src/coreclr/jit/emitarm64.cpp

```diff
@@ -11681,529 +11988,985 @@ void emitter::emitIns_R_R_R_R(instruction     ins,
     switch (ins)
     {
-        case INS_sve_ptrue:
-        case INS_sve_ptrues:
-            assert(isPredicateRegister(reg1));
-            assert(isScalableVectorSize(attr));
-            assert(insOptsScalableStandard(opt));
-            fmt = IF_SVE_DE_1A;
+        case INS_cset:
+        case INS_csetm:
+            assert(isGeneralRegister(reg));
+            cfi.cond = cond;
+            fmt      = IF_DR_1D;
             break;
 
         default:
             unreached();
             break;
 
     } // end switch (ins)
+
     assert(fmt != IF_NONE);
+    assert(isValidImmCond(cfi.immCFVal));
 
-    instrDesc* id = emitNewInstr(attr);
+    instrDesc* id = emitNewInstrSC(attr, cfi.immCFVal);
 
     id->idIns(ins);
     id->idInsFmt(fmt);
+    id->idInsOpt(INS_OPTS_NONE);
 
-    id->idReg1(reg1);
-    id->idInsOpt(opt);
-    id->idSvePattern(pattern);
+    id->idReg1(reg);
 
     dispIns(id);
     appendToCurIG(id);
 }
 
 /*****************************************************************************
  *
- *  Add an instruction referencing a register, a SVE Pattern and an immediate.
+ *  Add an instruction referencing two registers and a condition code
  */
 
-void emitter::emitIns_R_PATTERN_I(instruction ins, emitAttr attr, regNumber reg1, insSvePattern pattern, int imm)
+void emitter::emitIns_R_R_COND(instruction ins, emitAttr attr, regNumber reg1, regNumber reg2, insCond cond)
 {
-    emitAttr  size     = EA_SIZE(attr);
-    emitAttr  elemsize = EA_UNKNOWN;
-    insFormat fmt      = IF_NONE;
+    insFormat    fmt = IF_NONE;
+    condFlagsImm cfi;
+    cfi.immCFVal = 0;
 
     /* Figure out the encoding format of the instruction */
     switch (ins)
     {
-        case INS_sve_cntb:
-        case INS_sve_cntd:
-        case INS_sve_cnth:
-        case INS_sve_cntw:
+        case INS_cinc:
+        case INS_cinv:
+        case INS_cneg:
             assert(isGeneralRegister(reg1));
-            assert(size == EA_8BYTE);
-            assert(isValidUimm4From1(imm));
-            fmt = IF_SVE_BL_1A;
+            assert(isGeneralRegisterOrZR(reg2));
+            cfi.cond = cond;
+            fmt      = IF_DR_2D;
             break;
-
         default:
             unreached();
             break;
 
     } // end switch (ins)
+
     assert(fmt != IF_NONE);
+    assert(isValidImmCond(cfi.immCFVal));
 
-    instrDesc* id = emitNewInstrCns(attr, imm);
+    instrDesc* id = emitNewInstrSC(attr, cfi.immCFVal);
 
     id->idIns(ins);
     id->idInsFmt(fmt);
+    id->idInsOpt(INS_OPTS_NONE);
 
     id->idReg1(reg1);
-    id->idSvePattern(pattern);
+    id->idReg2(reg2);
 
     dispIns(id);
     appendToCurIG(id);
 }
 
 /*****************************************************************************
  *
- *  Add a memory barrier instruction with a 'barrier' immediate
+ *  Add an instruction referencing two registers and a condition code
  */
 
-void emitter::emitIns_BARR(instruction ins, insBarrier barrier)
```

Comment:
Am I right in thinking you've not touched `emitIns_BARR()` or `emitIns_R_R_R_COND()` and this is just the diff getting confused?

Which `emitIns_` functions have you changed?

Reply:
The diff is confused... never touched those. I only touched `emitIns_R_R_R_R`.

Reply:
I'm probably going to create a new `emitIns` called `emitInsSve_R_R_R_R` and put all the handling of the sve instructions in that and have `emitIns_R_R_R_R` call into it as a fallback. I'm hopeful that would limit the TP regressions and have it be more organized.